crates/bifrost/src/providers/replicated_loglet/tasks/find_tail.rs

```diff
@@ -0,0 +1,645 @@
+                                Some(Ok((node_id, tail_update))) = inflight_tail_update_watches.join_next() => {
+                                    // maybe sealed, maybe global is updated. Who knows.
+                                    nodeset_checker.merge_attribute(node_id, tail_update);
+                                    true
+                                }
+                                Some(Ok((node_id, info))) = inflight_info_requests.join_next() => {
+                                    nodeset_checker.merge_attribute(node_id, info);
+                                    true
+                                }
+                                else =>  false,
+                            };
+
+                            if nodeset_checker.any(NodeTailStatus::is_known_sealed) {
+                                // A node got sealed. Let's re-evaluate to finish up this seal before
+                                // we respond. We expect to switching to `SomeSealed`
+                                inflight_tail_update_watches.abort_all();
+                                continue 'check_nodeset;
+                            }
+
+                            if self.known_global_tail.latest_offset() >= max_local_tail {
+                                return FindTailResult::Open {
+                                    global_tail: self.known_global_tail.latest_offset(),
+                                };
+                            }
+                            // Is there a full write-quorum for the requested tail?
+                            if nodeset_checker.check_write_quorum(|attribute| match attribute {
+                                // We have all copies of the max-local-tail replicated. It's safe to
+                                // consider this offset as committed.
+                                NodeTailStatus::Known { local_tail, sealed } => {
+                                    // not sealed, and reached its local tail reached the max we
+                                    // are looking for.
+                                    !sealed && (*local_tail >= max_local_tail)
+                                }
+                                _ => false,
+                            }) {
+                                return FindTailResult::Open {
+                                    global_tail: max_local_tail,
+                                };
+                            }
+                            if !updated {
+                                // Nothing left to wait on. We tail didn't reach expected
+                                // target and no more nodes are expected to send us responses.
+                                return FindTailResult::Error(format!(
+                                    "Could not determine a safe tail offset for loglet_id={}, perhaps too many nodes down?",
+                                    self.my_params.loglet_id));
+                            }
```

Comment:
I am wondering if this check can happen immediately after the `tokio::select!` block? for clarity

Reply:
I think this changes the intention. The check is intentionally done after we exhaust the other options above.